src/routes/_surfaces/workspace/explore/filters/Filters.svelte

```diff
@@ -1,35 +1,124 @@
+  import {
+    clearSelectedDimensionLeaderboardAndUpdate,
+    clearSelectedLeaderboardValuesAndUpdate,
+    toggleSelectedLeaderboardValueAndUpdate,
+  } from "$lib/redux-store/explore/explore-apis";
   import { store } from "$lib/redux-store/store-root";
   import { isAnythingSelected } from "$lib/util/isAnythingSelected";
+  import type { Readable } from "svelte/store";
   import { fly } from "svelte/transition";
   export let metricsDefId;
   export let values;
 
-  function clearAllFilters() {
-    clearSelectedLeaderboardValuesAndUpdate(store.dispatch, metricsDefId);
+  let dimensions: Readable<DimensionDefinitionEntity[]>;
+  $: dimensions = getDimensionsByMetricsId(metricsDefId);
+
+  function clearFilterForDimension(dimension) {
+    clearSelectedDimensionLeaderboardAndUpdate(
+      store.dispatch,
+      metricsDefId,
+      dimension
+    );
   }
 
   $: hasFilters = isAnythingSelected(values);
+
+  function clearAllFilters() {
+    if (hasFilters)
+      clearSelectedLeaderboardValuesAndUpdate(store.dispatch, metricsDefId);
+  }
+
+  function pruneValues(set) {
+    if (!set) return;
+    return Object.keys(set)
+      .filter((key) => set[key].length)
+      .map((key) => {
+        return [key, set[key].filter(([_, v]) => v === true).map(([k]) => k)];
+      });
+  }
+
+  $: prunedValues = pruneValues(values);
+
+  function onSelectItem(event, item) {
+    toggleSelectedLeaderboardValueAndUpdate(
+      store.dispatch,
+      metricsDefId,
+      item.dimensionId,
+      event.detail,
+      true
+    );
+  }
 </script>
 
-<div class="pt-3 pb-3" style:min-height="50px">
-  {#if hasFilters}
-    <button
-      transition:fly|local={{ duration: 200, y: 5 }}
-      on:click={clearAllFilters}
-      class="
-            grid gap-x-2 items-center font-bold
-            bg-red-100
-            text-red-900
-            p-1
-            pl-2 pr-2
-            rounded
-        "
-      style:grid-template-columns="auto max-content"
+<section
+  class="pt-3 pb-3 grid gap-x-2"
+  style:grid-template-columns="max-content auto"
+  style:min-height="44px"
+>
+  <Tooltip
+    location="right"
+    alignment="middle"
+    distance={8}
+    suppress={!hasFilters}
+  >
+    <IconButton disabled={!hasFilters} on:click={clearAllFilters}>
+      <Filter />
+    </IconButton>
+
+    <TooltipContent slot="tooltip-content">
+      <TooltipShortcutContainer padTop>
+        <div>clear all filters</div>
+        <Shortcut><ShiftKey /> + Click</Shortcut>
+      </TooltipShortcutContainer>
+    </TooltipContent>
+  </Tooltip>
+  {#if prunedValues?.length && $dimensions?.length}
+    <ChipContainer>
+      {#each prunedValues as [dimensionId, selectedValues] (dimensionId)}
+        {@const dimension = $dimensions.find((dim) => dim.id === dimensionId)}
+        {@const name = dimension?.labelSingle?.length
+          ? dimension?.labelSingle
+          : dimension?.dimensionColumn}
+        <div animate:flip={{ duration: 200 }}>
```

Comment:
I think this section is a little harder to read than it needs to be-- how about constructing an array that contains only the data needed for the display the code section above. That'd simplify the `{#if prunedValues?.length && $dimensions?.length}` to one check and the two `@const`s could be removed here (clarify focus on the rendering)